In [0]:
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import confusion_matrix

In [0]:
# train from example 03
X_train_3 = pd.read_csv("example03-data.csv",header=None)
y_train_3 = pd.read_csv("example03-class.csv", header=None) 

In [0]:
# example 04
X_train_4 = pd.read_csv("example04-data.csv", header=None)
y_train_4 = pd.read_csv("example04-class.csv", header=None)

In [0]:
# last class, p mac's question
# ajarn instructs us to use all of example 4 to train
# and test with 50% of example 4 for both

# the second model is being tested with seen data

In [0]:
# 50% of example 4 as test
# splitting half
X_train, X_test, y_train, y_test = train_test_split(
    X_train_4, y_train_4, test_size=0.5, random_state=42)

In [0]:
# make array

# b - eg3
X_train_3 = np.asarray(X_train_3)
y_train_3 = np.asarray(y_train_3)

# c - eg4
X_train_4 = np.asarray(X_train_4)
y_train_4 = np.asarray(y_train_4)

# shared - 50% of eg4
X_test = np.asarray(X_test)
y_test = np.asarray(y_test)

In [0]:
# random forest model
# same model parameters on both datasets
model = RandomForestClassifier(n_estimators=100, random_state=100, oob_score=True, max_features=5)

##### (B)

In [37]:
forest_3 = model.fit(X_train_3, y_train_3)

y_pred_3 = forest_3.predict(X_test)
confusion = confusion_matrix(y_test,y_pred_3) 
print('Confusion Matrix : \n', confusion)

Confusion Matrix : 
 [[1 1]
 [1 2]]


/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:1: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  """Entry point for launching an IPython kernel.


In [38]:
# Plot confusion matrix-3
cm3 = confusion_matrix(y_test,y_pred_3, labels=[1,-1])

total = len(y_test)

# from confusion matrix calculate accuracy
accuracy=(cm3[0,0]+cm3[1,1])/total
print ('Accuracy : ', accuracy)

sensitivity = cm3[0,0]/(cm3[0,0]+cm3[0,1])
print('Sensitivity : ', sensitivity)

specificity = cm3[1,1]/(cm3[1,0]+cm3[1,1])
print('Specificity : ', specificity)

Accuracy :  0.6
Sensitivity :  0.6666666666666666
Specificity :  0.5


##### (C)

In [39]:
forest_4 = model.fit(X_train_4, y_train_4)

y_pred_4 = forest_4.predict(X_test)
confusion = confusion_matrix(y_test,y_pred_4) 
print('Confusion Matrix : \n', confusion)

Confusion Matrix : 
 [[2 0]
 [0 3]]


/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:1: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  """Entry point for launching an IPython kernel.


In [40]:
# Plot confusion matrix-4
cm4 = confusion_matrix(y_test,y_pred_4, labels=[1,-1])

total = len(y_test)

# from confusion matrix calculate accuracy
accuracy=(cm4[0,0]+cm4[1,1])/total
print ('Accuracy : ', accuracy)

sensitivity = cm4[0,0]/(cm4[0,0]+cm4[0,1])
print('Sensitivity : ', sensitivity)

specificity = cm4[1,1]/(cm4[1,0]+cm4[1,1])
print('Specificity : ', specificity)

Accuracy :  1.0
Sensitivity :  1.0
Specificity :  1.0


Assignment | Train Set | Train Size | Accuracy | Sensitivity | Specificity
-|-|-|-|-|-
 B | example03 | 10 | 0.60 | 0.67 | 0.5
 C | example04 | 10 | 1.0 | 1.0 | 1.0